src/dialog.scss

```diff
@@ -0,0 +1,170 @@
+
+  // RESIZE HANDLE
+  $fd-dialog-resize-handle-color: var(--sapButton_Lite_TextColor);
+
+  // BODY
+  $fd-dialog-body-padding-y: 1rem;
+  $fd-dialog-body-color: var(--sapGroup_ContentBackground);
+
+  // LOADER
+  $fd-dialog-loader-margin-y: 1.5rem;
+
+  // Footer
+  $fd-dialog-footer-button-min-width: 4rem;
+
+  display: none;
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background-color: $fd-dialog-overlay-color;
+
+  &--active {
+    display: block;
+  }
+
+  &__content {
+    @include fd-reset();
+
+    display: flex;
+    flex-direction: column;
+    overflow: hidden;
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    min-width: $fd-dialog-content-min-width;
+    min-height: $fd-dialog-content-min-height;
+    max-width: calc(100vw - #{$fd-dialog-content-margin-x});
+    max-height: calc(100vh - #{$fd-dialog-content-margin-y});
+    box-shadow: $fd-dialog-content-box-shadow;
+    border-radius: $fd-dialog-content-border-radius;
+
+    @each $size-label, $size in $fd-dialog-content-padding-x {
+      &--#{$size-label} {
+        .#{$block}__header,
+        .#{$block}__subheader,
+        .#{$block}__body,
+        .#{$block}__footer {
+          padding-left: map_get($fd-dialog-content-padding-x, $size-label);
+          padding-right: map_get($fd-dialog-content-padding-x, $size-label);
+        }
+      }
+    }
+
+    &--mobile {
+      width: 100vw;
+      height: 100vh;
+      max-width: 100vw;
+      max-height: 100vh;
+    }
+
+    &--no-mobile-stretch {
+      max-width: calc(100vw - #{$fd-dialog-content-margin-x});
+      max-height: calc(100vh - #{$fd-dialog-content-margin-y});
+    }
+
+    &--resizable {
+      @include fd-icon("resize-corner", "m", "after") {
+        color: $fd-dialog-resize-handle-color;
+        position: absolute;
+        cursor: nwse-resize;
+        right: 0;
+        bottom: -1px;
+      }
+    }
+
+    &--draggable {
+      &-grab {
+        @include fd-hover() {
+          .#{$block}__header {
+            cursor: grab;
+          }
+        }
+      }
+
+      &-grabbing {
+        @include fd-hover() {
+          .#{$block}__header {
+            cursor: grabbing;
+          }
+        }
+      }
+    }
+  }
+
+  &__header,
+  &__subheader,
+  &__footer {
+    padding-left: map_get($fd-dialog-content-padding-x, def);
+    padding-right: map_get($fd-dialog-content-padding-x, def);
+    flex-shrink: 0;
+  }
+
+  &__header {
+    .#{$block}__title {
+      @include fd-reset();
+      @include fd-ellipsis();
+    }
+  }
+
+  &__body {
+    @include fd-reset();
+
+    flex-grow: 1;
+    background-color: $fd-dialog-body-color;
+    padding-top: $fd-dialog-body-padding-y;
+    padding-bottom: $fd-dialog-body-padding-y;
+
+    &--no-vertical-padding {
+      padding-top: 0;
+      padding-bottom: 0;
+    }
+  }
+
+  &__footer {
+    .#{$block}__decisive-button {
+      min-width: $fd-dialog-footer-button-min-width;
+    }
```

Comment:
Since it's a requirement for all buttons inside Popover and Dialogs maybe it's better to add this requirement in the Bar component. And to "force" this rule we can target directly button elements. Because if the user omits this class the button won't be forced to apply the 4rem min width. Just a suggestion :)

Reply:
The thing is this should be applied only to "Begin" and "End" buttons. Docs are written as the suggest there can be other buttons inside footer which shouldn't be modified.

![image](https://user-images.githubusercontent.com/17496353/74824000-e7f9b680-5307-11ea-9ec9-aa0c287e870e.png)

Reply:
We can always target the first and the last child. My only concern is that the user can easily omit this class and to be honest it's easy to avoid this class by targeting the first and the last child of the footer/floating footer. Up to you to decide :)